tests/main/interfaces-system-dbus/task.yaml

```diff
@@ -0,0 +1,37 @@
+summary: Ensure that the dbus interface works on the system bus.
+
+execute: |
+    #shellcheck source=tests/lib/dirs.sh
+    . "$TESTSLIB/dirs.sh"
+
+    echo "Install dbus system test snaps"
+    snap install --edge test-snapd-dbus-provider
+    snap install --edge test-snapd-dbus-consumer
+
+    echo "Wait for dbus service to auto-start"
+    for _ in $(seq 60); do
+        if dbus-send --system --print-reply --dest=com.dbustest.HelloWorld /com/dbustest/HelloWorld com.dbustest.HelloWorld.SayHello | MATCH "hello world"; then
+            break
+        fi
+        sleep 1
+    done
+
+    echo "The dbus consumer plug is disconnected by default"
+    snap interfaces -i dbus | MATCH '^- +test-snapd-dbus-consumer:dbus-system-test'
+
+    if [ "$(snap debug confinement)" = partial ]; then
```

Comment:
I think we should exit after check the interface works when the plug is connected, otherwise we are not testing almost nothing on systems with no strict confinement.

Reply:
The stes we use are:
check default connection
connect if needed
test connected behabiour
check confinement and exit is needed
check disconnected behabiour

Reply:
TBH I just used the existing test (interfaces-dbus) as a blueprint. We should probably update this as well.

Reply:
@mvo5, interfaces-joystick is a nice example
Thanks

Reply:
> check confinement and exit is needed

We could execute a bunch of interface tests even if confinement is `partial`. Not this one in particular, as it requires apparmor dbus support, but a deal of other tests use apparmor features which are available outside of Ubuntu too, eg. on Arch and hopefully on openSUSE Leap 15.0 once it lands.